### PR TITLE
refactor: update `math/base/special/cexp` to follow latest project conventions

### LINKED ISSUE #3393: [RFC]: refactor `math/base/special/cexp` to follow latest project conventions
### Description

This RFC proposes refactoring of `math/base/special/cexp` to use in house stdlib functions for operations like `exp`, `sin`, `cos` and `copysign`.

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by aayush0325**
Resolves #3393

## Description

> What is the purpose of this pull request?

This pull request:

-   refactors `math/base/special/cexp` to follow latest project conventions to use functions developed by stdlib instead of the native C functions

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3393
## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by aayush0325** [comment]
good afternoon, please give this a review @gunjjoshi!

File: lib/node_modules/@stdlib/math/base/special/cexp/src/main.c [hunk 1]
```diff
@@ -17,6 +17,14 @@
...
+#include "stdlib/math/base/special/sin.h"
+#include "stdlib/math/base/special/cos.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/assert/is_infinite.h"
+#include "stdlib/constants/float64/pinf.h"
+#include "stdlib/math/base/special/copysign.h"
+#include "stdlib/constants/float64/ninf.h"
 #include "stdlib/complex/float64/ctor.h"
 #include "stdlib/complex/float64/reim.h"
 #include <math.h>
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/cexp/src/main.c]:
```suggestion
```

Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/cexp/src/main.c]: We can remove this header now.

**Comment by gunjjoshi** [comment]
@aayush0325 Can you replace the use of `stdlib_base_sin` and `stdlib_base_cos` with `stdlib_base_sincos` at [`math/base/special/sincos`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/sincos)? That would match what we have in the existing JS implementation of `cexp` here: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/cexp/lib/main.js#L101

**Comment by aayush0325** [comment]
i've pushed the requested changes @gunjjoshi

**Review by gunjjoshi — APPROVED** [review]

**Pull request merged**